### PR TITLE
add touch method

### PR DESCRIPTION
**Pull request by TheOldMop**

### CODE REVIEW

**Comment by jdufresne** [comment]
Could this be done using the native `TOUCH` command? It would avoid all race conditions between `.get()` and `.set()`.

https://redis.io/commands/touch

**Comment by TheOldMop** [comment]
No.
For django `touch` set new expiration time for key:
`cache.touch('a', 10)`
`True`
For redis `touch`  alters the last access time of a key(s)

They have different meaning.

**Comment by jdufresne** [comment]
Thanks.  Maybe we could use `EXPIRE` or `EXPIREAT` then. They don't have a race condition and can work on existent or non-existent keys.

```
127.0.0.1:6379> expire mykey 10
(integer) 0
127.0.0.1:6379> set mykey myvalue
OK
127.0.0.1:6379> expire mykey 10
(integer) 1
```

**Comment by TheOldMop** [comment]
I missed this method ( Can fix it tomorrow

File: django_redis/cache.py [hunk 1]
```diff
@@ -150,3 +150,14 @@ def lock(self, *args, **kwargs):
     @omit_exception
     def close(self, **kwargs):
         self.client.close(**kwargs)
+
+    @omit_exception
+    def touch(self, key, timeout=None, version=None):
+        try:
+            return self.client.touch(key, timeout=timeout, version=version)
+        except ConnectionInterrupted as e:
```
Comment (jdufresne) [inline django_redis/cache.py]: Isn't this exception already handled the `omit_exception`? Why does it need to be repeated?

File: django_redis/client/default.py [hunk 2]
```diff
@@ -540,3 +540,19 @@ def close(self, **kwargs):
...
+        """
+        Sets a new expiration for a key.
+        """
+
+        if client is None:
+            client = self.get_client(write=True)
+
+        key = self.make_key(key, version=version)
+
+        if client.exists(key):
```
Comment (jdufresne) [inline django_redis/client/default.py]: Can we drop this check? My understanding is `expire` will ignore keys that doesn't exist.

Could use a test for the key-doesn't-exist case.

File: tests/test_backend.py [hunk 3]
```diff
@@ -635,6 +635,30 @@ def test_master_slave_switching(self):
         except NotImplementedError:
             pass
 
+    def test_touch_zero_timeout(self):
+        self.cache.set("test_key", 222, timeout=10)
+
+        self.cache.touch("test_key", 0)
```
Comment (jdufresne) [inline tests/test_backend.py]: I think it would be useful to assert the return value of `touch`:

```py
self.assertIs(res, True)
```

**Comment by niwinz** [comment]
Thanks!

**Comment by niwinz** [comment]
Seems like some tests fails on master related to touch method, can you check this please?

**Comment by TheOldMop** [comment]
I can check but only around 20:00 UTC +3

**Pull request merged**